pkg/arvo/sys/vane/ames.hoon

```diff
@@ -2566,8 +2595,29 @@
       %hear  (on-hear [message-num fragment-num]:task)
       %done  (on-done message-num.task)
       %wake  (on-wake current.task)
+      %prod  on-prod
       %halt  set-wake
     ==
+  ::  +on-prod: reset congestion control, re-send packets
+  ::
+  ++  on-prod
+    ^+  packet-pump
+    ?:  =(~ next-wake.state)
```

Comment:
Why is this not just `?~  next-wake.state`

Reply:
because of hoon's TMI problem -- if you do `?~`, then within the expressions that comes after it, the type of the argument is ~ in one branch, and the non-null type in the other branch (in this case, `_[~ u=*@da]`) .  This is fine unless you want to "reassign" a new value of the other subtype to .next-state later on, which will nest-fail.  This function does this by calling the +give function and overwriting the whole |packet-pump core with its result, which might contain either a null or non-null .next-wake.

So instead, we can use `=(~ next-wake)`, which does not specialize the type and thereby avoids the TMI problem.

There's been discussion of adding a rune that would be like `=.` but would allow an LHS field's type to be expanded, which would obviate this kind of TMI, since assignment would just expand the .next-wake in the subject back to either null or non-null instead of nest-fail'ing.